bobcat/bb-cumber/src/main/java/com/cognifide/qa/bb/cumber/Bobcumber.java

```diff
@@ -80,66 +86,45 @@
    */
   public Bobcumber(Class<?> clazz) throws InitializationError, IOException {
     super(clazz);
+    bindFields();
     storeFailedResults = clazz.isAnnotationPresent(StoreFailedResults.class);
     if (storeFailedResults) {
-      file = new File(clazz.getAnnotation(StoreFailedResults.class).value());
-      if (!file.exists()) {
-        PrintWriter writer = new PrintWriter(file, CharEncoding.UTF_8);
-        writer.close();
-      }
-      if (RERUN_FAILED_TESTS_CLASS_NAME.equals(clazz.getName()) && isFeatureFileEmpty()) {
-        noTestsToRerun = true;
-      }
+      featureFile = createFile(clazz.getAnnotation(StoreFailedResults.class).value());
+      statisticsFile = createFile(statisticsFilePath);
     }
+    isItFailedTestsRerun = clazz.isAnnotationPresent(FailedTestsRunner.class);
   }
 
   @Override
   public void run(RunNotifier notifier) {
-    if (noTestsToRerun) {
-      notifier.fireTestFinished(Description.EMPTY);
-      return;
+    if (isItFailedTestsRerun) {
+      try {
+        if (StatisticsUtils.getNumberOfFailedTests(statisticsFile).equals(0)) {
+          notifier.fireTestFinished(Description.EMPTY);
+          return;
+        }
+        if (StatisticsUtils.getPercentageOfFailedTests(statisticsFile) > maxFailedTestPercentage) {
+          notifier.fireTestFailure(new Failure(Description.createSuiteDescription("Too many tests."), new TooManyTestsException()));
```

Comment:
Both `Too many tests` and `TooManyTestsException` names are not properly describing problem